internal/domain/domain_operations.go

```diff
@@ -0,0 +1,242 @@
+			_, ok := clusterIdsInTheCurrentDomain[clusterObj.ID]
+			// go over clusters that are in the domain, skip the rest
+			if !ok {
+				continue
+			}
+			if domainCluster["name"] == clusterObj.Name {
+				domainCluster["id"] = clusterObj.ID
+				domainCluster["primary_datastore_name"] = clusterObj.PrimaryDatastoreName
+				domainCluster["primary_datastore_type"] = clusterObj.PrimaryDatastoreType
+				domainCluster["is_default"] = clusterObj.IsDefault
+				domainCluster["is_stretched"] = clusterObj.IsStretched
+			}
+		}
+	}
+	_ = data.Set("cluster", domainClusterData)
+
+	return nil
+}
+
+func SetBasicDomainAttributes(ctx context.Context, domainId string, data *schema.ResourceData,
+	apiClient *client.VcfClient) (*models.Domain, error) {
+	getDomainParams := domains.NewGetDomainParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+	getDomainParams.ID = domainId
+	domainResult, err := apiClient.Domains.GetDomain(getDomainParams)
+	if err != nil {
+		return nil, err
+	}
+	domain := domainResult.Payload
+
+	data.SetId(domain.ID)
+	_ = data.Set("name", domain.Name)
+	_ = data.Set("status", domain.Status)
+	_ = data.Set("type", domain.Type)
+	_ = data.Set("sso_id", domain.SSOID)
+	_ = data.Set("sso_name", domain.SSOName)
+	_ = data.Set("is_management_sso_domain", domain.IsManagementSSODomain)
+	if len(domain.VCENTERS) < 1 {
+		return nil, fmt.Errorf("no vCenter Server instance found for domain %q", domainId)
+	}
+	vcenterConfigAttribute, vcenterConfigExists := data.GetOk("vcenter")
+	var vcenterConfigRaw []interface{}
+	if vcenterConfigExists {
+		vcenterConfigRaw = vcenterConfigAttribute.([]interface{})
+	} else {
+		vcenterConfigRaw = *new([]interface{})
+		vcenterConfigRaw = append(vcenterConfigRaw, make(map[string]interface{}))
+	}
+	vcenterConfig := vcenterConfigRaw[0].(map[string]interface{})
+	vcenterConfig["id"] = domain.VCENTERS[0].ID
+	vcenterConfig["fqdn"] = domain.VCENTERS[0].Fqdn
+	_ = data.Set("vcenter", vcenterConfigRaw)
+
+	return domain, nil
+}
+
+func CreateDomainUpdateSpec(data *schema.ResourceData, markForDeletion bool) *models.DomainUpdateSpec {
+	result := new(models.DomainUpdateSpec)
+	if markForDeletion {
+		result.MarkForDeletion = true
+		return result
+	}
+	if data.HasChange("name") {
+		result.Name = data.Get("name").(string)
+	}
+
+	// TODO implement support for IPPoolSpecs in NsxTSpec
+	// by placing the added cluster spec in the DomainUpdateSpec
+	//nsxtSpec, err := generateNsxSpecFromResourceData(data)
+	//if err == nil {
+	//	result.NsxTSpec = nsxtSpec
+	//} else {
+	//	return nil, err
+	//}
+
+	return result
+}
+
+func ImportDomain(ctx context.Context, data *schema.ResourceData, apiClient *client.VcfClient,
+	domainId string, allowManagementDomain bool) ([]*schema.ResourceData, error) {
+	domainObj, err := SetBasicDomainAttributes(ctx, domainId, data, apiClient)
+	if err != nil {
+		return nil, err
+	}
+	if !allowManagementDomain && data.Get("type").(string) == "MANAGEMENT" {
```

Comment:
Why do we have this restriction? 

Note that many customers use just the management domain and either run workloads in the management cluster (cluster 0) or create additional clusters in the management domain.

Reply:
So that customers won't be able to import their management domain and later (accidentaly) delete it.

If they want to add another cluster to the management domain, they can use a domain datasource pointing to the management domain, and create separate cluster (as a separate resource, rather than subresource), that has "domain_id" pointint go the ID ot the management domain